src/test/java/org/datadog/jmxfetch/TestGCMetrics.java

```diff
@@ -0,0 +1,182 @@
+            final List<Map<String, Object>> actualMetrics = startAngGetMetrics(server, true);
+            assertThat(actualMetrics, hasSize(13));
+            assertGCMetric(actualMetrics, "jvm.gc.minor_collection_count", "ParNew", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.minor_collection_time", "ParNew", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.major_collection_count", "ConcurrentMarkSweep", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.major_collection_time", "ConcurrentMarkSweep", "counter");
+        }
+    }
+
+    @Test
+    public void testDefaultNewGCMetricsUseG1GC() throws IOException {
+        try (final MisbehavingJMXServer server = new MisbehavingJMXServer(
+                MisbehavingJMXServer.JDK_17,
+                "-XX:+UseG1GC -Xmx128M -Xms128M")) {
+            final List<Map<String, Object>> actualMetrics = startAngGetMetrics(server, true);
+            assertThat(actualMetrics, hasSize(13));
+            assertGCMetric(actualMetrics, "jvm.gc.minor_collection_count", "G1 Young Generation", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.minor_collection_time", "G1 Young Generation", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.major_collection_count", "G1 Old Generation", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.major_collection_time", "G1 Old Generation", "counter");
+        }
+    }
+
+    @Test
+    public void testDefaultNewGCMetricsUseZGC() throws IOException {
+        try (final MisbehavingJMXServer server = new MisbehavingJMXServer(
+                MisbehavingJMXServer.JDK_17,
+                "-XX:+UseZGC -Xmx128M -Xms128M")) {
+            final List<Map<String, Object>> actualMetrics = startAngGetMetrics(server, true);
+            assertThat(actualMetrics, hasSize(13));
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_pauses_collection_count", "ZGC Pauses", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_pauses_collection_time", "ZGC Pauses", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_cycles_collection_count", "ZGC Cycles", "counter");
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_cycles_collection_time", "ZGC Cycles", "counter");
+        }
+    }
+
+    private List<Map<String, Object>> startAngGetMetrics(final MisbehavingJMXServer server, final boolean newGCMetrics) throws IOException {
+        server.start();
+        this.initApplicationWithYamlLines(
+            "init_config:",
+            "  is_jmx: true",
+            "  new_gc_metrics: " + newGCMetrics,
+            "",
+            "instances:",
+            "    -   name: jmxint_container",
+            "        host: " + server.getIp(),
+            "        collect_default_jvm_metrics: true",
+            "        max_returned_metrics: 300000",
+            "        port: " + server.getRMIPort());
+        // Run one iteration first
```

Comment:
I'd still like to understand this better. Should we create a follow-up card to investigate this?
If we're hitting this case in this test, it could be something customers experience as well